stackslib/src/chainstate/stacks/boot/pox-4.clar

```diff
@@ -912,19 +921,25 @@
           (asserts! (is-eq pox-addr (get pox-addr existing-entry))
                     (err ERR_DELEGATION_WRONG_REWARD_SLOT))
 
+          ;; Validate that amount is less than or equal to `max-amount`
+          (asserts! (>= max-amount increased-entry-total) (err ERR_SIGNER_AUTH_AMOUNT_TOO_HIGH))
+
+          ;; Verify signature from delegate that allows this sender for this cycle
+          ;; 'lock-period' param set to one period, same as aggregation-commit-indexed
+          (try! (consume-signer-key-authorization pox-addr reward-cycle "agg-increase" u1 signer-sig signer-key increased-entry-total max-amount auth-id))
+
           ;; update the pox-address list -- bump the total-ustx
           (map-set reward-cycle-pox-address-list
                    { reward-cycle: reward-cycle, index: reward-cycle-index }
                    { pox-addr: pox-addr,
-                     total-ustx: increased-ustx,
+                     total-ustx: increased-entry-total,
                      stacker: none,
-                     ;; TODO: this must be authorized with a signature, or tx-sender allowance!
-                     signer: (get signer existing-entry) })
+                     signer: signer-key })
```

Comment:
You need to check that this is the same signer key in `existing-entry`, no?  `stack-increase` does this by way of `increase-reward-cycle-entry`.

Reply:
Ah-- good catch. I thought allowing this to change was okay, but actually, yes, we must enforce equality here, because increasing an existing entry isn't otherwise auth-checked: anyone can choose to increase a given entry. But they shouldn't be allowed to change the signer key.

Reply:
Copy. Changing this accordingly.

Reply:
Changed.